### PR TITLE
update openssl version

### PR DESCRIPTION
**Pull request by omarudolley**
- Create a new Ubuntu image with updated OpenSSL 
- Extend base images from the new ubuntu image

### CODE REVIEW

File: nginx-base/alpine-nginx/Dockerfile [hunk 1]
```diff
@@ -5,8 +5,8 @@ ENV PARSE_TEMPLATE_VERSION=v1.0.0 \
 
 RUN set -exu \
  # fixing curl versions there to fix CVE-2021-22945
- && apk add --no-cache --virtual build-deps curl~=7.79.1-r0 \
- && apk add --no-cache libcurl~=7.79.1-r0 xz-libs~=5.2.5-r1 xz~=5.2.5-r1 \
+ && apk add --no-cache --virtual build-deps curl~=7.83 \
```
Comment (lietu) [inline nginx-base/alpine-nginx/Dockerfile:8]: Why are any of these version numbers listed?

File: ubuntu/ubuntu20.04/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM ubuntu:20.04
+
+RUN : \
+ && set -exu \
+ && apt update\
```
Comment (lietu) [inline ubuntu/ubuntu20.04/Dockerfile:5]: `apt` does not provide a stable API for scripting

Comment (lietu) [reply ubuntu/ubuntu20.04/Dockerfile:5]: Did you check if this actually does anything, is the base image updated?

File: nodejs-base/ubuntu20.04-node14/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:20.04
+FROM ubuntu:ubuntu20.04
```
Comment (lietu) [inline nodejs-base/ubuntu20.04-node14/Dockerfile:1]: What does this do?

File: python-base/ubuntu20.04-pyston2.2/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:20.04
+FROM ubuntu:ubuntu20.04
```
Comment (lietu) [inline python-base/ubuntu20.04-pyston2.2/Dockerfile:1]: What does this do?

File: python-base/ubuntu20.04-python3.9/Dockerfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:20.04
+FROM ubuntu:ubuntu20.04
```
Comment (lietu) [inline python-base/ubuntu20.04-python3.9/Dockerfile:1]: What does this do?

**Review by lietu — CHANGES_REQUESTED** [review]

**Comment by lietu** [comment]
https://hub.docker.com/_/ubuntu/?tab=tags&page=1&name=ubuntu20.04

File: python-base/ubuntu20.04-python3.9/Dockerfile (hunk 5 above)
Comment (omarudolley) [reply python-base/ubuntu20.04-python3.9/Dockerfile:1]: I created a new ubuntu image with an updated `OpenSSL` and I am now extending the base images from the new ubuntu image.

File: python-base/ubuntu20.04-pyston2.2/Dockerfile (hunk 4 above)
Comment (omarudolley) [reply python-base/ubuntu20.04-pyston2.2/Dockerfile:1]: Created a new ubuntu image with an updated `OpenSSL` -- now extending the base images from the new ubuntu image.

File: nodejs-base/ubuntu20.04-node14/Dockerfile (hunk 3 above)
Comment (omarudolley) [reply nodejs-base/ubuntu20.04-node14/Dockerfile:1]: Created a new ubuntu image with an updated OpenSSL -- now extending the base images from the new ubuntu image.

Comment (lietu) [reply nodejs-base/ubuntu20.04-node14/Dockerfile:1]: No it's not.

File: python-base/ubuntu20.04-pyston2.2/Dockerfile (hunk 4 above)
Comment (lietu) [reply python-base/ubuntu20.04-pyston2.2/Dockerfile:1]: No it's not.

File: python-base/ubuntu20.04-python3.9/Dockerfile (hunk 5 above)
Comment (lietu) [reply python-base/ubuntu20.04-python3.9/Dockerfile:1]: No it's not.

**Comment by lietu** [comment]
Did you actually run the build for these locally?

File: nodejs-base/ubuntu20.04-node14/Dockerfile (hunk 3 above)
Comment (lietu) [reply nodejs-base/ubuntu20.04-node14/Dockerfile:1]: Or rather, it's doing that accidentally. The point is, you're now reusing a public name, `ubuntu`, this does not get published in the docker hub for `ubuntu`. This is very unclear and very prone to cause issues.

File: ubuntu/ubuntu20.04/Dockerfile (hunk 2 above)
Comment (omarudolley) [reply ubuntu/ubuntu20.04/Dockerfile:5]: I guess not. The version before and after the update is 
`OpenSSL 1.1.1f  31 Mar 2020` 
As I understand the CVE https://www.openssl.org/news/secadv/20220705.txt is patched in v1.1.1f
https://launchpad.net/ubuntu/+source/openssl/1.1.1f-1ubuntu2.16 

So does this mean that there is no need to update?

Comment (omarudolley) [reply ubuntu/ubuntu20.04/Dockerfile:5]: @lietu

Comment (lietu) [reply ubuntu/ubuntu20.04/Dockerfile:5]: https://computingforgeeks.com/check-version-of-installed-package-on-ubuntu-debian/

Comment (omarudolley) [reply ubuntu/ubuntu20.04/Dockerfile:5]: I checked the version with `dpkg` and I end up with the same result. 

```
# dpkg -s openssl | grep Version                             
Version: 1.1.1f-1ubuntu2.16
# 
```

**Pull request closed without merging**